compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -894,6 +894,23 @@ void addLiteralValue(DiagnosticPos pos, Set<Whitespace> ws, int typeTag, Object
         addExpressionNode(litExpr);
     }
 
+    void addByteArrayLiteralExprNode(DiagnosticPos pos, Set<Whitespace> ws, int typeTag, byte[] blob) {
+        BLangArrayLiteral arrayLiteral = (BLangArrayLiteral) TreeBuilder.createArrayLiteralNode();
+        List<ExpressionNode> byteArrayLiteral = new ArrayList<>();
+        for (byte b : blob) {
```

Comment:
We need to preserve original expression here. Don't desguar until desugar phase.